File: ir/augassign_ir.py
from .identified_ir_node import IdentifiedIRNode, IRNode
from common.span import SourceSpan
from common.operators import Operator


class AugAssignIR(IRNode):
    def __int__(
        self, target: str, op: Operator, value=IRNode, span: SourceSpan = None
    ):
        super().__init__(span=span)
        self.span=span
        self.target=target
        self.op=op
        self.value=value

File: rust/src/ir/nodes/augassign_ir.rs
use crate::ir::{ExprIR, Operator, SourceSpan};

#[derive(Debug, Clone)]
pub struct AugAssignIR {
    // Python dump annotates target as str, but to_proto calls target.to_proto(),
    // so this is modeled as an expression target.
    pub target: Box<ExprIR>,
    pub op: Operator,
    pub value: Box<ExprIR>,
    pub span: Option<SourceSpan>,
}

impl AugAssignIR {
    pub fn new(target: ExprIR, op: Operator, value: ExprIR, span: Option<SourceSpan>) -> Self {
        Self {
            target: Box::new(target),
            op,
            value: Box::new(value),
            span,
        }
    }
}
